File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        exit(1);
    }

    int k = atoi(argv[1]);

    if (k < 0)
    {
        printf("Usage: ./caesar k\n");
        exit(1);
    }

    string s = get_string("plaintext: ");

    int i, n;

    char cipher[strlen(s) + 1];
    for (i = 0, n = strlen(s); i < n; i++)
    {
        if (islower(s[i]))
        {
            cipher[i] = (char)((((int) s[i] + k - 97) % 26) + 97);
        }

        else if (isupper(s[i]))
        {
            cipher[i] = (char)((((int) s[i] + k - 65) % 26) + 65);
        }

        else
        {
            cipher[i] = s[i];
        }
    }
    printf("ciphertext: %s\n", cipher);
}

File: pset6/sentimental/caesar/caesar.py
from sys import argv
from sys import exit
from cs50 import get_string


def main():
    if len(argv) != 2:
        exit("Usage: python caesar.py k")
    k = int(argv[1])
    if k < 0:
        exit("Usage: python caesar.py k")
    plaintext = get_string("plaintext: ")

    cipher = []
    for c in plaintext:
        if c.islower():
            cipher.append(chr(((ord(c) + k - 97) % 26) + 97))
        elif c.isupper():
            cipher.append(chr(((ord(c) + k - 65) % 26) + 65))
        else:
            cipher.append(c)
    ciphertext = ''.join(cipher)
    print('ciphertext:', ciphertext)


if __name__ == "__main__":
    main()